www/js/control/AppStatusModal.tsx

```diff
@@ -0,0 +1,450 @@
+
+        let overallFitnessName = t('intro.appstatus.overall-fitness-name-android');
+        let locExplanation = t('intro.appstatus.overall-loc-description');
+        if(platform == "ios") {
+            overallFitnessName = t('intro.appstatus.overall-fitness-name-ios');
+            if(osver < 13) {
+                locExplanation = (t("intro.permissions.locationPermExplanation-ios-lt-13"));
+            } else {
+                locExplanation = (t("intro.permissions.locationPermExplanation-ios-gte-13"));
+            }
+        }
+        tempExplanations.push({name: t('intro.appstatus.overall-loc-name'), desc: locExplanation});
+        tempExplanations.push({name: overallFitnessName, desc: t('intro.appstatus.overall-fitness-description')});
+        tempExplanations.push({name: t('intro.appstatus.overall-notification-name'), desc: t('intro.appstatus.overall-notification-description')});
+        tempExplanations.push({name: t('intro.appstatus.overall-background-restrictions-name'), desc: t('intro.appstatus.overall-background-restrictions-description')});
+
+        setExplanationList(tempExplanations);
+  
+        //waiting on samsung feedback, need more information
+        setBackgroundRestricted(false);
+        if(window['device'].manufacturer.toLowerCase() == "samsung") {
+          setBackgroundRestricted(true);
+          setAllowBackgroundInstructions(t("intro.allow_background.samsung"));
+        }
+  
+        console.log("Explanation = "+explanationList);
+    }
+
+    function createChecklist(){
+        if(platform == "android") {
+            setupAndroidLocChecks();
+            setupAndroidFitnessChecks();
+            setupAndroidNotificationChecks();
+            setupAndroidBackgroundRestrictionChecks();
+        } else if (platform == "ios") {
+            setupIOSLocChecks();
+            setupIOSFitnessChecks();
+            setupAndroidNotificationChecks();
+        } else {
+            setError("Alert! unknownplatform, no tracking");
+            setErrorVis(true);
+            console.log("Alert! unknownplatform, no tracking"); //need an alert, can use AlertBar?
+        }
+        
+        refreshAllChecks();
+    }
+
+    //refreshing checks with the plugins to update the check's statusState
+    function refreshAllChecks() {
+        //refresh each check
+        checkList.forEach((lc) => {
+            lc.refresh();
+        });
+        console.log("setting checks are", checkList);
+        recomputeOverallStatus();
+    }
+
+    //recomputing checks updates the visual cues of their status
+    function recomputeAllChecks() {
+        console.log("recomputing checks", checkList);
+        checkList.forEach((lc) => {
+            lc.statusIcon = iconMap(lc.statusState);
+            lc.statusColor = colorMap(lc.statusState)
+        });
+        recomputeOverallStatus();
+    }
+
+    //custom hook executes function on app resuming foreground
+    useAppStateChange( function() {
```

Comment:
This is the code responsible for launching the model if a permission is wrong. I just noticed that it also launches when the permissions are correct, so I am working to debug and find a fix.

Reply:
I think, by refactoring a bit, I've resolved this issue. Tested performance by closing and reopening app with good permissions, logging out and logging back in (had issues showing modal before consent at one point), and leaving and returning to app (background -> foreground) without the modal popping up. Also left, turned off location permission, and reopened, which did launch the modal. 

Closing the app, breaking a permission, and re-opening the app opened the modal only when I navigated to the profile page. If someone is using the app as directed, this wouldn't really ever happen, so I'm not sure it's a big concern.